File: Web/src/domain/scores/tunes/neseHaliaVodu.ts
import type { Score, ScoreEvent } from '../score.js'
import { blow, draw, rest } from '../scoreWriting.js'

const carryingTheWater: ScoreEvent[] = [
    draw(4, 1), draw(4, 1), draw(5, 1),
    draw(5, 1), draw(6, 2, { vibrato: 0.6 }), draw(6, 1),
    blow(6, 1), draw(5, 1), blow(5, 1),
    draw(4, 2, { vibrato: 0.6 })
]

const ivankoFollowing: ScoreEvent[] = [
    draw(4, 1), draw(4, 1), draw(5, 1),
    draw(5, 1), draw(6, 2, { vibrato: 0.6 }), draw(6, 1),
    blow(6, 1), draw(5, 1), blow(5, 1),
    draw(4, 2, { vibrato: 0.6 })
]

export const neseHaliaVodu: Score = {
    name: 'Nese Halia vodu',
    key: 'A',
    position: 'third',
    beatsPerMinute: 80,
    events: [...carryingTheWater, rest(1), ...ivankoFollowing]
}

File: Core/Sources/HarmonicaCore/Entities/Scores/NeseHaliaVodu.swift
public extension Score {
    static let neseHaliaVodu = Score(
        name: "Nese Halia vodu",
        key: .a,
        position: .third,
        beatsPerMinute: 80,
        events: carryingTheWater + [.rest(beats: 1)] + ivankoFollowing
    )

    private static let carryingTheWater: [ScoreEvent] = [
        .draw(.four, 1), .draw(.four, 1), .draw(.five, 1),
        .draw(.five, 1), .draw(.six, 2, vibrato: 0.6), .draw(.six, 1),
        .blow(.six, 1), .draw(.five, 1), .blow(.five, 1),
        .draw(.four, 2, vibrato: 0.6)
    ]

    private static let ivankoFollowing: [ScoreEvent] = [
        .draw(.four, 1), .draw(.four, 1), .draw(.five, 1),
        .draw(.five, 1), .draw(.six, 2, vibrato: 0.6), .draw(.six, 1),
        .blow(.six, 1), .draw(.five, 1), .blow(.five, 1),
        .draw(.four, 2, vibrato: 0.6)
    ]
}
